junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -71,11 +74,13 @@ protected void invokeTestMethod(JupiterEngineExecutionContext context, DynamicTe
 			Object instance = testExtensionContext.getTestInstance();
 			Object testFactoryMethodResult = executableInvoker.invoke(getTestMethod(), instance, testExtensionContext,
 				context.getExtensionRegistry());
-
-			try (Stream<DynamicTest> dynamicTestStream = toDynamicTestStream(testFactoryMethodResult)) {
+			TestSource source = getSource().orElseThrow(AssertionError::new);
```

Comment:
I know this "cannot happen", but I'd be a little more comfortable with `JUnitException::new`.